src/main/java/org/elasticsearch/rest/action/admin/indices/analyze/RestAnalyzeAction.java

```diff
@@ -47,21 +58,72 @@ public RestAnalyzeAction(Settings settings, RestController controller, Client cl
     @Override
     public void handleRequest(final RestRequest request, final RestChannel channel, final Client client) {
         String text = request.param("text");
+        AnalyzeRequest analyzeRequest = new AnalyzeRequest(request.param("index"), text);
+
         if (text == null && request.hasContent()) {
-            text = request.content().toUtf8();
-        }
-        if (text == null) {
-            throw new ElasticsearchIllegalArgumentException("text is missing");
+            XContentType type = XContentFactory.xContentType(request.content());
+            if (type == null) {
+                text = request.content().toUtf8();
+            } else {
+                // NOTE: if rest request with xcontent body has request parameters, these parameters override xcontent values
+                analyzeRequest = buildFromContent(request.content(), analyzeRequest);
+            }
         }
 
-        AnalyzeRequest analyzeRequest = new AnalyzeRequest(request.param("index"), text);
         analyzeRequest.listenerThreaded(false);
+        analyzeRequest.text(text);
         analyzeRequest.preferLocal(request.paramAsBoolean("prefer_local", analyzeRequest.preferLocalShard()));
         analyzeRequest.analyzer(request.param("analyzer"));
         analyzeRequest.field(request.param("field"));
         analyzeRequest.tokenizer(request.param("tokenizer"));
         analyzeRequest.tokenFilters(request.paramAsStringArray("token_filters", request.paramAsStringArray("filters", analyzeRequest.tokenFilters())));
         analyzeRequest.charFilters(request.paramAsStringArray("char_filters", analyzeRequest.charFilters()));
+
+        if (analyzeRequest.text() == null) {
```

Comment:
I think this is a duplicate of org.elasticsearch.action.admin.indices.analyze.AnalyzeRequest#validate and can be removed?